rollup.config.js

```diff
@@ -11,55 +10,67 @@ const exportedComponents = fs
     fd => fd.isDirectory() && fd.name !== 'common' && fd.name !== '_internal'
   )
   .map(({ name }) => name);
+const external = [
+  ...Object.keys(pkg.dependencies || {}),
+  ...Object.keys(pkg.peerDependencies || {}),
+  ...Object.keys(pkg.optionalDependencies || {}),
+];
+const typescriptOpts = {
+  tsconfigOverride: {
+    exclude: [
+      "dist",
+      "**/*.test.*",
+      "**/*.stories.*",
+      "**/mocks"
+    ],
+  }
+};
 
 export default [
   {
-    external: [
-      ...Object.keys(pkg.dependencies || {}),
-      ...Object.keys(pkg.peerDependencies || {}),
-    ],
-    input: 'src/index.ts',
+    external,
+    context: 'window',
+    input: {
+      'index.esm': 'src/index.ts',
+      ...exportedComponents.reduce((acc, component) => {
+        acc[
+          `components/${component}/index`
+          ] = `src/components/${component}/index.ts`;
+        return acc;
+      }, {}),
+    },
     output: [
       {
-        file: pkg.main,
-        format: 'cjs',
-      },
-      {
-        file: pkg.module,
-        format: 'es',
+        dir: 'dist',
+        entryFileNames: '[name].js',
+        chunkFileNames: 'chunks/chunk-[hash].js',
+        exports: 'named',
+        format: 'esm',
       },
     ],
     plugins: [
       typescript({
-        typescript: require('typescript'),
-        exclude: ['**/*.(test|stories).ts+(|x)', '**/mocks/*.ts+(|x)'],
+        ...typescriptOpts
       }),
-      json(),
     ],
   },
   {
-    external: [
-      ...Object.keys(pkg.dependencies || {}),
-      ...Object.keys(pkg.peerDependencies || {}),
+    external,
+    input: 'src/index.ts',
+    context: 'window',
+    output:[
+      {
+        file: pkg.main,
+        format: 'cjs',
+      },
     ],
-    input: exportedComponents.reduce((acc, component) => {
-      acc[
-        `components/${component}/index`
-      ] = `src/components/${component}/index.ts`;
-      return acc;
-    }, {}),
-    output: {
-      dir: 'dist',
-      entryFileNames: '[name].js',
-      chunkFileNames: 'chunks/chunk-[hash].js',
-      format: 'cjs',
-    },
     plugins: [
       typescript({
-        typescript: require('typescript'),
-        exclude: ['**/*.(test|stories).ts+(|x)', '**/mocks/*.ts+(|x)'],
+        ...typescriptOpts,
+        compilerOptions: {
+          declaration: false,
+        },
```

Comment:
I am not familiar with rollup configs, do you know what you are doing here?)

Reply:
yes, I'm disabling re-generation declaration files here, cause it was generated on the previous step, it will speed up bundling process